packages/DateInput/src/DateInput.js

```diff
@@ -0,0 +1,186 @@
+import React from "react";
+import PropTypes from "prop-types";
+import moment from "moment";
+import CalendarIcon from "@paprika/icon/lib/Calendar";
+import Input from "@paprika/input";
+import { isMomentObjectOrNull } from "@paprika/helpers/lib/customPropTypes";
+import useI18n from "@paprika/l10n/lib/useI18n";
+
+const INPUT_PARSE_ERROR = "INPUT_PARSE";
+
+const propTypes = {
+  /** If the value of <input> is valid or not. */
+  hasError: PropTypes.bool,
+
+  /** Date format used while entering and parsing user input. */
+  dateFormat: PropTypes.string,
+
+  /** Selected date in moment object. */
+  date: isMomentObjectOrNull,
+
+  /** Possible date in moment object. This prop is used when user is typing date and we can't determinate exact date. */
+  possibleDate: isMomentObjectOrNull,
+
+  /** Date format used while displaying date. It should be human-friendly and spelled out, default is MMMM DD,YYYY */
+  humanFormat: PropTypes.string,
+
+  /** Callback when date is inputed. Will be called on blur or enter key press. */
+  onChange: PropTypes.func,
+
+  /** Callback when user inputs date. Will be called after every key up event. */
+  onChangePossibleDate: PropTypes.func,
+
+  /** Error callback. Will be called on blur or enter key press if inputted date can't be parsed. */
+  onError: PropTypes.func,
+
+  onClick: PropTypes.func,
+
+  /** Guard function. If it returns true - confirmation will be prevented. */
+  denyConfirmation: PropTypes.func,
+
+  /** Callback when confirm */
+  beforeConfirmation: PropTypes.func,
+};
+
+const defaultProps = {
+  hasError: false,
+  dateFormat: "MM/DD/YYYY",
+  date: null,
+  possibleDate: null,
+  humanFormat: undefined,
```

Comment:
why cannot be null moment conflict?

Reply:
Well, if it was set to `null` then the default value (from i18n) would not be set:
https://github.com/acl-services/paprika/pull/457/files#diff-1f6beb3c48ccc9dff8f817d5d81f1732R67